src/bindings/python/flux/util.py

```diff
@@ -380,3 +382,163 @@ def format(self, obj):
+        "compact": Connectors("│ ", "├─", "└─", "  "),
+    }
+
+    def __init__(self, label, prefix="", combine_children=False):
+
+        self.label = label
+        self.prefix = prefix
+        if self.prefix:
+            self.prefix += " "
+
+        self.duplicates = {}
+        self.children = []
+        self.count = 1
+        self.combine_children = combine_children
+
+    def append(self, label, prefix=""):
+        """Append a new child, possibly combining duplicates
+
+        Args:
+            label: Label for this node
+            prefix: Optional prefix to display before tree part
+
+        """
+        if not self.combine_children:
+            self.add(label, prefix, combine_children=False)
+            return
+        clabel = f"{prefix}{label}"
+        if clabel in self.duplicates:
+            self.duplicates[clabel].increment()
+        else:
+            tree = self.add(label, prefix, combine_children=True)
+            self.duplicates[clabel] = tree
+
+    def append_tree(self, tree):
+        """Add a child Tree, combining duplicates
+
+        Args:
+            tree: the Tree object to append
+        """
+        if not self.combine_children or tree.children:
+            self.children.append(tree)
+            return
+        clabel = f"{tree.prefix}{tree.label}"
+        if clabel in self.duplicates:
+            self.duplicates[clabel].increment()
+        else:
+            self.children.append(tree)
+            self.duplicates[clabel] = tree
+
+    def add(self, label, prefix="", combine_children=False):
+        """Add a child by prefix and label, does not combine duplicates
+
+        Args:
+            label: Label for this node
+            prefix: Optional prefix to display before tree part
+            combine_children: True if like children should be combined
+
+        Returns:
+            Tree: the new child Tree object
+        """
+        tree = Tree(label, prefix=prefix, combine_children=combine_children)
+        self.children.append(tree)
+        return tree
+
+    def increment(self):
+        self.count = self.count + 1
+
+    def _render(
+        self,
+        pstack=None,
+        connector="",
+        style="box",
+        level=None,
+        truncate=None,
+    ):
+        if pstack is None:
+            pstack = []
+        if level and level < len(pstack):
+            return
+
+        if self.count > 1:
+            label = f"{self.count}*[{self.label}]"
+        else:
+            label = self.label
+
+        prefix = "".join(pstack[:-1])
+        result = f"{self.prefix}{prefix}{connector}{label}"
+        if truncate and len(result) > truncate:
+            result = result[: truncate - 1] + "+"
+        print(result)
+
+        connectors = self.connector_styles[style]
+        for child in self.children:
+            if child == self.children[-1]:
+                pstack.append(connectors.BLANK)
+                connector = connectors.ELBOW
```

Comment:
not sure if `connector` is the best variable since you have the `connectors` variable.  Although unsure of a better variable name.

Reply:
Ok, I'll try to think of something better.

Reply:
Changed `connectors` to `cstyle` in latest fixup commit.